### DISCUSSION TITLE
What does size attribute of number field type do?

### DISCUSSION THREAD

**Discussion by acalvino4** [craftcms/docs#440] [Ideas] [answered]
When I go to create a number field type in the craft cp, there is a 'size' field. What does this do?

![image](https://user-images.githubusercontent.com/43254861/201750525-3830939a-606f-4a28-af40-e3217d484746.png)

The [docs](https://craftcms.com/docs/4.x/number-fields.html#settings) say no more than "Size – The input’s size attribute."
Is this referring to the html [size attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/size)? This would seem odd, bc it is not offered for any other field, bc the craft section settings already allow setting the input width (as shown below), and especially bc the mdn docs say that `<input type=number>` [doesn't support](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#controlling_input_size) the size attribute.

![image](https://user-images.githubusercontent.com/43254861/201750398-41b11c57-f648-4f8d-aae9-98a6dddc62ca.png)

So what _does_ this do?

**Comment by AugustMiller** [comment, marked as answer]
Sorry for any confusion. This may just be a legacy feature that has persisted, but I think it makes some sense… here's a few tidbits that may clarify what's going on:

### Generalizations About Inputs

- Many inputs in the control panel are allowed to be the full width of their parent.
- Number inputs, however, don't often need very many characters—their values will probably share a general "magnitude" or number of significant places (either as large values, or values with a specific number of decimal places).

### Input Type

The "Number" field is still rendered as an `<input type="text">` element, in the Craft control panel:

```html
<input
  type="text"
  id="fields-rating"
  class="text"
  name="fields[rating][value]"
  size="42"
  inputmode="numeric"
  autocomplete="off"
  dir="ltr">
```

☝️ Note the [`inputmode` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode), which is responsible for `type="number"`-esque UX improvements (like preventing inadvertent alphabetic input from causing validation issues).

### Size Attribute

This _is_ added—and does indeed work—because the generated HTML uses the `text` type.

---

Hope this helps!
